addons/mail/models/res_partner.py

```diff
@@ -35,7 +36,17 @@ def _compute_contact_address_inline(self):
             partner.contact_address_inline = re.sub(r'\n(\s|\n)*', ', ', partner.contact_address).strip().strip(',')
 
     def _compute_im_status(self):
-        super()._compute_im_status()
+        status_by_partner = {}
+        for presence in self.env["mail.presence"].search([("user_id", "in", self.user_ids.ids)]):
+            partner = presence.user_id.partner_id
+            if (
+                status_by_partner.get(partner, "offline") == "offline"
+                or presence.status == "online"
+            ):
+                status_by_partner[partner] = presence.status
+        for partner in self:
+            default_status = "offline" if partner.user_ids else "im_partner"
```

Comment:
I wonder if default status could be put in the line below. The variable is not necessary IMO.